src/docfx/build/dependency/DependencyResolver.cs

```diff
@@ -224,7 +224,14 @@ public string GetRelativeUrl(Document fileRelativeTo, Document file)
 
                     var file = Document.TryCreateFromFile(relativeTo.Docset, pathToDocset);
 
-                    return (file != null ? null : Errors.FileNotFound(relativeTo.ToString(), path), file, null, query, fragment, null, pathToDocset);
+                    // try to resolve with .md for landing page
+                    if (file is null && forLandingPage)
```

Comment:
just a suggestion, move this out of resolveLink method, hack it in caller.

Reply:
Added a comment, this flag should not be used. And the hack needs to be removed once the user input is correct.